### PR TITLE
Add ingest pipeline to include OPSI data

### PR DESCRIPTION
**Pull request by rbayliss**
This PR is the piece that's necessary to add OPSI data to the licensing API. It:

* Adds an additional [Python Lambda function](https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/licensing-ingest-prod-ingest/versions/$LATEST?tab=graph) to handle reaching out to the Lightsail server to pick up new data files for each agency.  This will replace the existing cron scripts that are in place on the Lightsail server. It should be fine to run both the cron scripts and the lambda at the same time - they shouldn't collide in any way. We can tear down the cron scripts once we're up and running.
* Adds ETL processing of OPSI data so that it is included in the database.
* Adds a list of agencies (`agencies.json`) that defines where we keep data for each agency, and how that agency's data should be processed.
* Changes the license details API to allow querying for OPSI licenses (`dept_name = 'OPSI'`).
* Changes the license details API to return OPSI licenses when you query for EOPSS licenses.  This is a temporary BC layer so that existing API consumers do not need to change their implementation overnight.

*Steps to Test*:
For testing, this work has been deployed to our dev environment.

* Using the attached Postman profile, create a new "Environment" for our dev environment using the following values:
    * `baseUrl`: https://dev.licensing.api.digital.mass.gov/v1
    * `apiKey`: Our standard testing API key, which is `Q8jA35Ekw785DW7jPreqedvOb1wgadx2k1xbP9De`
    * `username`: Your username (must be created at https://licensing.api.digital.mass.gov/register.html first)
    * `password`: Your password
* Use the "GetLicense" request to fetch license data on license `AI-119025` (`{{baseUrl}}/licenses/AI-119025`).  Verify that two records appear (one will be from from OPSI, one from EOPSS).
* Use the "GetLicenseByAgency" request to fetch license data for `OPSI` license `AI-119025`.  There should be one license returned, because you are querying only the OPSI dataset.
* Use the "GetLicenseByAgency" request to fetch license data for `EOPSS` license `AI-119025`.  There should be two license returned, because you are querying both the OPSI and EOPSS datasets (before we launch this change, EOPSS will drop OPSI licenses from their dataset).

To test the pipeline process:
* Invoke the [ingest lambda function](https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/licensing-ingest-prod-ingest?tab=graph) and verify that it runs without errors.
* Look into the [S3 bucket's `output` folder for OPSI](https://s3.console.aws.amazon.com/s3/buckets/api-upload.digital.mass.gov/data/output/OPSI/?region=us-east-1&tab=overview) and verify that you see a file from at least yesterday.

### CODE REVIEW

**Comment by rbayliss** [comment]
Youssef, assigning this to you for review on the Python bit. Please don't merge this though - it needs to be merged on a schedule.

**Comment by rbayliss** [comment]
And the postman profile.  You will need to rename it so it's a `.json` file before importing into postman (github doesn't allow JSON uploads).
[postman.txt](https://github.com/massgov/mds-licensing/files/2713263/postman.txt)

File: ingest/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  pythonRequirements:
+    dockerizePip: true
+
+functions:
+  ingest:
+    handler: ingest.handler
+    timeout: 900
+    enabled: true
+    # Run at the top of the hour, every hour.
+    rate: "cron(0 * * * ? *)"
```
Comment (jlandfried) [inline ingest/serverless.yml]: I _think_ on the dev environment based on this. Looking at the syntax though, I think we need to nest this under "events" -> "schedule" like this:

```
...
events:
      - schedule:
          rate: "cron(0 * * * ? *)"
...
```

Looking at the monitoring for the lambda, it does not look like it is running on a schedule, but this makes me think that it should be. Unless there is something preventing it from running for development.

File: ingest/ingest.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Handler for moving data from Lightsail SFTP instance to S3.
+"""
+
+import io
+import json
+import boto3
+import paramiko
+
+with open('agencies.json', 'r') as f:
+  agencies = json.load(f)
```
Comment (yriahi) [inline ingest/ingest.py]: All code below need to be indented (same as agencies line)
`with open...` will self close the open file at the end.
so while the file is open, all other operations/function need to be indented.

Another way, we can have:
```
my_file = open('agencies.json', 'r')
agencies = json.load(my_file)
...code...
...code...
my_file.close()
```

File: ingest/serverless.yml (hunk 1 above)
Comment (rbayliss) [reply ingest/serverless.yml]: Good call!  Pushing a fix for this now.  Thanks!

File: ingest/convert.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+"""
+def get_agency(input_bucket, input_key):
+  uri = f"s3://{input_bucket}/{input_key}"
+  for agency in agencies:
+    if "s3" in agency and agency['s3']['input'] in uri:
+      return agency
+
+  raise Error(f"Unable to determine the agency for {uri}")
+
+with open('sample.json', 'r') as f:
```
Comment (yriahi) [inline ingest/convert.py]: Looks like a test file; but I believe this will be removed in a subsequent commit?

**Comment by rbayliss** [comment]
Closing as it's been moved into #30.

**Pull request closed without merging**